pkg/openwsn/contrib/leds.c

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (C) 2017 Hamburg University of Applied Sciences
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_openwsn
+ * @{
+ *
+ * @file
+ *
+ * @author      Michael Frey <michael.frey@msasafety.com>
+ * @author      Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include "leds.h"
+#include "openwsn_leds.h"
+
+#ifdef MODULE_ZTIMER_USEC
+#include "ztimer.h"
+#endif
+#include "board.h"
+#include "periph/gpio.h"
+
+#include <stdint.h>
+#include <string.h>
+
+/** holds the internal configuration for debug pins */
+static ledpins_config_t configuration = {
+    GPIO_UNDEF,
+    GPIO_UNDEF,
+    GPIO_UNDEF,
+    GPIO_UNDEF,
+    GPIO_LED_HIGH,
+};
+
+void ledpins_riot_init(const ledpins_config_t *user_config)
+{
+    if (IS_USED(MODULE_OPENWSN_LEDPINS)) {
+        if (user_config != NULL) {
+            memcpy(&configuration, user_config, sizeof(ledpins_config_t));
+            leds_init();
+        }
+    }
+    else {
+        (void)user_config;
+    }
+}
+
+void leds_init(void)
+{
+    if (IS_USED(MODULE_OPENWSN_LEDPINS)) {
+        if (configuration.error != GPIO_UNDEF) {
+            gpio_init(configuration.error, GPIO_OUT);
+        }
+        else if (configuration.sync != GPIO_UNDEF) {
+            gpio_init(configuration.sync, GPIO_OUT);
+        }
+        else if (configuration.radio != GPIO_UNDEF) {
+            gpio_init(configuration.radio, GPIO_OUT);
+        }
+        else if (configuration.debug != GPIO_UNDEF) {
+            gpio_init(configuration.debug, GPIO_OUT);
+        }
+        leds_all_off();
```

Comment:
Just saw this, should not be else if. Also I'll apply Kaspar suggestion for the debugpins here as well.

Reply:
done.